Commands/Files/AddFile.cs

```diff
@@ -124,6 +132,47 @@ protected override void ExecuteCmdlet()
 
                 ClientContext.ExecuteQueryRetry();
             }
+            if (ContentType != null)
+            {
+                ClientContext.Load(file, f=>f.ListId);
+                ClientContext.ExecuteQueryRetry();
+
+                List list = SelectedWeb.Lists.GetById(file.ListId);
+                ClientContext.Load(list);
+                ClientContext.ExecuteQueryRetry();
+
+                ContentType ct;
+                if (!string.IsNullOrEmpty(ContentType.Id))
+                {
+                    ct = list.GetContentTypeById(ContentType.Id);
+                }
+                else if(!string.IsNullOrEmpty(ContentType.Name))
+                {
+                    ct = list.GetContentTypeByName(ContentType.Name);
+                }
+                else if(ContentType.ContentType != null)
+                {
+                    ct = ContentType.ContentType;
+                }
+                else
+                {
+                    ct = null;
+                   
+                }
+
+                if (ct != null)
+                {
+                    var item = file.ListItemAllFields;
+                    item["ContentTypeId"] = ct.Id.StringValue;
+                    item.Update();
+                    ClientContext.ExecuteQueryRetry();
+                }
+                else
+                {
+                    WriteVerbose("Content Type was 'null' was not found.");
```

Comment:
I would now swallow this. They explicitly state they want to set the content type, if that fails because the CT doesn't exist they will not be notified which can have all kinds of side effects with incorrect assignments of content types (as it falls back to the default content type).